compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -8219,4 +8229,14 @@ private FieldInfo(List<BType> types, boolean required, boolean readonly) {
             this.readonly = readonly;
         }
     }
+
+    private static class TypeSymbolPair {
```

Comment:
Should we document what this is for, since this is required for somewhat of a special case?